### PR TITLE
Add ability to define a message to be displayed at package install time

### LINKED ISSUE #47: Add an optional message which is displayed when a package is installed
Packages should be able to specify a message that is displayed to the user when the package is installed.

### PR DESCRIPTION
**Pull request by mapitman**
Fixes #47

### CODE REVIEW

**Comment by mapitman** [comment]
If you decide to accept this, I'll send a PR for the packages I've added that need install messages.

File: cmd/install.go [hunk 1]
```diff
@@ -72,6 +72,9 @@ var installCommand = &cobra.Command{
 			return err
 		}
 		fmt.Printf("🐳  Installed %s to %s\n", imageName, path.Join(pm.InstallPath, pkg.Name))
+		if len(pkg.InstallMessage) > 0 {
```
Comment (themalkolm) [inline cmd/install.go]: Maybe check with default value instead? Feels more idiomatic:

```
if pkg.InstallMessage != "" {
...
```

**Comment by themalkolm** [comment]
I'm not a committer here. Just want to keep up with the tool development.

So.. I find it confusing as `install-message` doesn't really tell me when this message will be shown. Will it be shown pre, during or post install?

@mapitman what do you think? 'post-install-message'?

**Comment by mapitman** [comment]
I like `post-install-message`. I also agree with your other comment about checking the value. I'll make those changes a bit later today.

File: packages/package.go [hunk 2]
```diff
@@ -60,8 +60,8 @@ func NewPackageFromImage(image string, imageInspect types.ImageInspect) (*Packag
 			}
 		}
 
-		if installMessage, ok := labels["io.whalebrew.install-message"]; ok {
-			pkg.InstallMessage = installMessage
+		if installMessage, ok := labels["io.whalebrew.post-install-message"]; ok {
```
Comment (themalkolm) [inline packages/package.go]: s/installMessage/postInstallMessage/ ?

File: packages/package_test.go [hunk 3]
```diff
@@ -23,16 +23,16 @@ func TestNewPackageFromImage(t *testing.T) {
 				"io.whalebrew.config.environment": "[\"SOME_CONFIG_OPTION\"]",
 				"io.whalebrew.config.volumes":     "[\"/somesource:/somedest\"]",
 				"io.whalebrew.config.ports":       "[\"8100:8100\"]",
-				"io.whalebrew.install-message":		"Some message to display during install",
+				"io.whalebrew.post-install-message":    "Some message to display during install",
```
Comment (themalkolm) [inline packages/package_test.go]: s/during install/after install/ ?

Comment (mapitman) [inline packages/package_test.go]: eh, yeah. It's just a test value, so not too big of a deal

**Comment by mapitman** [comment]
I squashed all these changes down to one commit

**Comment by mapitman** [comment]
@themalkolm thanks for the code review, this will help keep it clean!

**Comment by bfirsh** [comment]
Thanks both - this looks great!

@mapitman needs a rebase though sorry. :( Should have merged your PR first because it's been sitting around for a while.

**Comment by mapitman** [comment]
Fixed the conflicts

**Comment by bfirsh** [comment]
The one thing holding me back here is this is the first change that makes this file incompatible with the Compose file service format.

I don't think there's a good reason to make them completely compatible, but I'd at least like to make clear which options are used by Docker and which are used by Whalebrew. (E.g. by namespacing Whalebrew options under a `x-whalebrew` key). 

I think we can defer this problem entirely though, because the post install message doesn't even need to be written to the package if it's only used at install time, right? I'd suggest just removing it from the package entirely.

**Comment by mapitman** [comment]
@bfirsh I was a little confused about your comments at first, but now I think I got it. Are you saying we should remove the YAML serialization hints from the Package struct?

**Comment by bfirsh** [comment]
Yep! Or, perhaps remove it from the struct entirely.

**Comment by mapitman** [comment]
Probably not going to get the changes done for this, so closing.

**Pull request closed without merging**